cns/kubecontroller/nodenetworkconfig/reconciler.go

```diff
@@ -157,29 +158,32 @@ func (r *Reconciler) Started(ctx context.Context) (bool, error) {
 }
 
 // SetupWithManager Sets up the reconciler with a new manager, filtering using NodeNetworkConfigFilter on nodeName.
-func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, node *v1.Node) error {
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, node *v1.Node, cnsconfig *configuration.CNSConfig) error {
```

Comment:
instead of the whole cnsconfig, can we pass in a `checkGeneration bool` or something like that? don't want the nnc reconciler depending on the cns/config